motr/client.h

```diff
@@ -1750,11 +1827,44 @@ int m0_client_layout_capture(struct m0_client_layout *layout,
 		      struct m0_obj *obj,
 		      struct m0_client_layout **out);
 
+
 /* Allocate/free in-memory layout data struct for an object. */
 struct m0_client_layout*
 m0_client_layout_alloc(enum m0_client_layout_type type);
 void m0_client_layout_free(struct m0_client_layout *layout);
 
+/**
+ * Calculate checksum/protection info for data/KV
+ *
+ * @param pi  Caller will pass Generic pi struct, which will be typecasted to
+ *            specific PI type struct. API will calculate the checksum and set
+ *            pi_value of PI type struct. In case of digest, caller will send
+ *            data unit N-1 unit's digest via prev_digest field in PI type struct.
+ *            This api will calculate unit N's digest and set value in curr_digest.
+ * @param seed seed value (obj_id+data_unit_offset) required to calculate
+ *             the checksum. If this pointer is NULL that means either 
+ *             this checksum calculation is meant for KV or user does
+ *             not want seeding.
+ * @param m0_bufvec - Set of buffers for which checksum is computed. Normally
+ *                    this set of vectors will make one data unit.
+ * @param flag if flag is M0_PI_CALC_UNIT_ZERO, it means this api is called for 
+ *             first data unit and init functionality should be invoked such as MD5_Init.
+ * @param[out] curr_digest digest of data unit N, will be required to calculate checksum for
+ *                         next data unit, N+1. This api will calculate and set value for this field.
+ *                         NOTE: curr_digest always have unseeded digest.
```

Comment:
You can add that this parameter is mandatory

Reply:
Done.